### ISSUE TITLE
ERROR: Could not resolve "web-worker"

### ISSUE THREAD

**Issue by vincerubinetti** [cytoscape/cytoscape.js-elk#48]
Getting this error trying to use this package with a modern Vite SPA with ESBuild:

```
✘ [ERROR] Could not resolve "web-worker"

    node_modules/elkjs/lib/main.js:38:29:
      38 │         var Worker = require('web-worker');
         ╵                              ~~~~~~~~~~~~

  You can mark the path "web-worker" as external to exclude it from the bundle, which will remove
  this error and leave the unresolved path in the bundle. You can also surround this "require" call
  with a try/catch block to handle this failure at run-time instead of bundle-time.

/Users/vincerubinetti/Desktop/molevolvr2.0/frontend/node_modules/esbuild/lib/main.js:1472
  let error = new Error(text);
              ^

Error: Build failed with 1 error:
node_modules/elkjs/lib/main.js:38:29: ERROR: Could not resolve "web-worker"
    at failureErrorWithLog (/Users/vincerubinetti/Desktop/molevolvr2.0/frontend/node_modules/esbuild/lib/main.js:1472:15)
    at /Users/vincerubinetti/Desktop/molevolvr2.0/frontend/node_modules/esbuild/lib/main.js:945:25
    at /Users/vincerubinetti/Desktop/molevolvr2.0/frontend/node_modules/esbuild/lib/main.js:1353:9
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  errors: [Getter/Setter],
  warnings: [Getter/Setter]
}

Node.js v22.5.0
error: script "dev" exited with code 1
```

**Comment by isYunchaoWang** [comment]
How to fix this issue?

**Comment by maxkfranz** [comment]
Fixed in 2.3
